packages/S11-modules/multi-version/lib/MultiVer.rakumod

```diff
@@ -0,0 +1,142 @@
+use v6;
+unit class MultiVer;
+use Test::Util;
+
+class Foo:ver<0.0.2> { }
+
+has $.prefix;
+has @.distributions;
+has $.standalone;
+has $.standalone-script;
+has $.inst-dir;
+has $.inst-repo;
+has $.use-lib;
+
+submethod TWEAK {
+    $!prefix //= make-temp-dir;
+}
+
+method mksubdir(IO:D() $path) {
+    my $fullpath = $path.is-relative ?? $!prefix.add($path) !! $path;
+    return $fullpath if $fullpath.e;
+    my @path = $*SPEC.splitdir: $fullpath.resolve.relative($!prefix).IO;
+    my $subdir = $!prefix;
+    for @path -> $dir {
+        die "Can only create dirs under \$.prefix, but '" ~ $path ~ "' is outside of it" if $dir eq '..';
+        $subdir .= add($dir);
+        next if $subdir.e;
+        $subdir.mkdir;
+    }
+    $subdir
+}
+
+method rmsubdir(IO:D() $entry) {
+    return unless $entry.e;
+    if $entry.d {
+        for $entry.dir -> $subentry {
+            next if $subentry.basename eq '..' | '.';
+            if $subentry.d {
+                self.rmsubdir($subentry);
+            }
+            else {
+                $subentry.unlink;
+            }
+        }
+        $entry.rmdir;
+    }
+    else {
+        $entry.unink;
+    }
+}
+
+method make-dist(Str:D :$dist = 'MVDist', :%fields, Bool :$standalone, Bool :$include = True) {
```

Comment:
fwiw there is a similar utility function / pattern used elsewhere:
https://github.com/Raku/roast/blob/4ec7d433a47ee7344316959db9a0a6cdf90a918d/S11-repository/cur-candidates.t#L151-L159

Reply:
Actually I see this uses templates instead of generating everything

Reply:
I was considering it, but the particular needs of this test made it easier to implement own version.

Reply:
> Actually I see this uses templates instead of generating everything

Somehow github didn't show this comment to me earlier. 

Yes, I use templates. And, BTW, it reminds me that I forgot about doing a control test for module version.